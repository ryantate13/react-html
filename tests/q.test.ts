import {q} from '../src';

describe('q', () => {
    it('renders with children', () => {
        expect(q({}, 'TEST')).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(q({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(q()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(q({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(q({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(q({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(q({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(q({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(q({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(q({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(q({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(q({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(q({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(q({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(q({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(q({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(q({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(q({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden="true"', () => {
        expect(q({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden="false"', () => {
        expect(q({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(q({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(q({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(q({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(q({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(q({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(q({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(q({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(q({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(q({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(q({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(q({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(q({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(q({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope="true"', () => {
        expect(q({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope="false"', () => {
        expect(q({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(q({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(q({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(q({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(q({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(q({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(q({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style="[object Object]"', () => {
        expect(q({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex="1"', () => {
        expect(q({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(q({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(q({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(q({translate: "no"})).toMatchSnapshot();
    });

    it('renders with cite="https://google.com/"', () => {
        expect(q({cite: "https://google.com/"})).toMatchSnapshot();
    });
});