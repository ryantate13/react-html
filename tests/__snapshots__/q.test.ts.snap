// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`q renders arrays of children 1`] = `
<q>
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
</q>
`;

exports[`q renders with accessKey="a,b,c,d" 1`] = `
<q
  accessKey="a,b,c,d"
/>
`;

exports[`q renders with autoCapitalize="characters" 1`] = `
<q
  autoCapitalize="characters"
/>
`;

exports[`q renders with autoCapitalize="none" 1`] = `
<q
  autoCapitalize="none"
/>
`;

exports[`q renders with autoCapitalize="off" 1`] = `
<q
  autoCapitalize="off"
/>
`;

exports[`q renders with autoCapitalize="on" 1`] = `
<q
  autoCapitalize="on"
/>
`;

exports[`q renders with autoCapitalize="sentences" 1`] = `
<q
  autoCapitalize="sentences"
/>
`;

exports[`q renders with autoCapitalize="words" 1`] = `
<q
  autoCapitalize="words"
/>
`;

exports[`q renders with children 1`] = `
<q>
  TEST
</q>
`;

exports[`q renders with cite="https://google.com/" 1`] = `
<q
  cite="https://google.com/"
/>
`;

exports[`q renders with className="foo bar baz" 1`] = `
<q
  className="foo bar baz"
/>
`;

exports[`q renders with contentEditable="false" 1`] = `
<q
  contentEditable="false"
/>
`;

exports[`q renders with contentEditable="true" 1`] = `
<q
  contentEditable="true"
/>
`;

exports[`q renders with dir="auto" 1`] = `
<q
  dir="auto"
/>
`;

exports[`q renders with dir="ltr" 1`] = `
<q
  dir="ltr"
/>
`;

exports[`q renders with dir="rtl" 1`] = `
<q
  dir="rtl"
/>
`;

exports[`q renders with draggable="false" 1`] = `
<q
  draggable="false"
/>
`;

exports[`q renders with draggable="true" 1`] = `
<q
  draggable="true"
/>
`;

exports[`q renders with empty attributes 1`] = `<q />`;

exports[`q renders with hidden=false 1`] = `
<q
  hidden={false}
/>
`;

exports[`q renders with hidden=true 1`] = `
<q
  hidden={true}
/>
`;

exports[`q renders with id="TEST" 1`] = `
<q
  id="TEST"
/>
`;

exports[`q renders with inputMode="decimal" 1`] = `
<q
  inputMode="decimal"
/>
`;

exports[`q renders with inputMode="email" 1`] = `
<q
  inputMode="email"
/>
`;

exports[`q renders with inputMode="none" 1`] = `
<q
  inputMode="none"
/>
`;

exports[`q renders with inputMode="numeric" 1`] = `
<q
  inputMode="numeric"
/>
`;

exports[`q renders with inputMode="search" 1`] = `
<q
  inputMode="search"
/>
`;

exports[`q renders with inputMode="tel" 1`] = `
<q
  inputMode="tel"
/>
`;

exports[`q renders with inputMode="text" 1`] = `
<q
  inputMode="text"
/>
`;

exports[`q renders with inputMode="url" 1`] = `
<q
  inputMode="url"
/>
`;

exports[`q renders with is="foo" 1`] = `
<q
  is="foo"
/>
`;

exports[`q renders with itemID="https://google.com/" 1`] = `
<q
  itemID="https://google.com/"
/>
`;

exports[`q renders with itemProp="https://google.com/" 1`] = `
<q
  itemProp="https://google.com/"
/>
`;

exports[`q renders with itemRef="foo bar" 1`] = `
<q
  itemRef="foo bar"
/>
`;

exports[`q renders with itemScope=false 1`] = `
<q
  itemScope={false}
/>
`;

exports[`q renders with itemScope=true 1`] = `
<q
  itemScope={true}
/>
`;

exports[`q renders with itemType="https://google.com/" 1`] = `
<q
  itemType="https://google.com/"
/>
`;

exports[`q renders with lang="" 1`] = `
<q
  lang=""
/>
`;

exports[`q renders with lang="es-419" 1`] = `
<q
  lang="es-419"
/>
`;

exports[`q renders with slot="TEST" 1`] = `
<q
  slot="TEST"
/>
`;

exports[`q renders with spellCheck="false" 1`] = `
<q
  spellCheck="false"
/>
`;

exports[`q renders with spellCheck="true" 1`] = `
<q
  spellCheck="true"
/>
`;

exports[`q renders with style={"color":"#000000"} 1`] = `
<q
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`q renders with tabIndex=1 1`] = `
<q
  tabIndex={1}
/>
`;

exports[`q renders with title="TEST" 1`] = `
<q
  title="TEST"
/>
`;

exports[`q renders with translate="no" 1`] = `
<q
  translate="no"
/>
`;

exports[`q renders with translate="yes" 1`] = `
<q
  translate="yes"
/>
`;

exports[`q renders without attributes 1`] = `<q />`;
